src/main/java/manufacturing/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,55 @@
+package manufacturing.dao.impl;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import manufacturing.dao.ManufacturerDao;
+import manufacturing.db.Storage;
+import manufacturing.lib.Dao;
+import manufacturing.model.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> getById(Long manufacturerId) {
+        return Optional.ofNullable(Storage.manufacturers.get(manufacturerId.intValue()));
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer manufacturerForUpdate = Storage.manufacturers
+                .stream()
+                .filter(m -> Objects.equals(m.getId(), manufacturer.getId()))
+                .findFirst()
+                .get();
+        manufacturerForUpdate.setName(manufacturer.getName());
+        manufacturerForUpdate.setCountry(manufacturer.getCountry());
+        return manufacturerForUpdate;
+    }
+
+    @Override
+    public boolean deleteById(Long manufacturerId) {
+        for (Manufacturer manufacturerForDeletion : getAllManufacturers()) {
```

Comment:
it's better to use here removeIf method of the list

Reply:
done